patches/net/minecraft/client/Minecraft.java.patch

```diff
@@ -326,14 +326,16 @@
      }
  
      private boolean isLevelRunningNormally() {
-@@ -2056,6 +_,7 @@
+@@ -2056,7 +_,8 @@
      }
  
      public void setLevel(ClientLevel p_91157_, ReceivingLevelScreen.Reason p_341652_) {
+-        this.updateScreenAndTick(new ReceivingLevelScreen(() -> false, p_341652_));
 +        if (this.level != null) net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.event.level.LevelEvent.Unload(this.level));
-         this.updateScreenAndTick(new ReceivingLevelScreen(() -> false, p_341652_));
++        this.updateScreenAndTick(net.neoforged.neoforge.client.DimensionTransitionScreenManager.getScreen(this.level == null ? null : p_91157_.dimension(), this.level == null ? null : this.level.dimension()).create(() -> false, p_341652_));
```

Comment:
`this.level == null ? null : p_91157_.dimension()` looks wrong to me. "If the current dimension is null, don't pass the new dimension"? If that's the intended logic, I suggest either adding a comment or moving the "null out target dimension" to the receiving method---here it looks like a "don't call a method on a null value" check (which I strongly suspect it is).

Reply:
It is intended as the level will be null on login. Vanilla only plays the transition screens when actually travelling to dimensions, not logging into them, so I replicated that behavior here. I will add a comment for clarity if desired

Reply:
Yes, this needs a comment.

How about you change getScreen() to take two Nullable Levels and put the logic there? There you have all the space you want to write a comment, here in the patch isn't really the place for comments.

Reply:
The issue with changing it to use levels comes with the `ClientPacketListener` patch. `CommonPlayerSpawnInfo` doesn't provide a level, only a `ResourceKey`. That patch also checks that the player isn't dying so the transition screen doesn't display on respawn. I can still add the necessary comments in the `DimensionTransitionScreenManager` class, but I don't know if that will be as effective as adding the comment to the patch

Reply:
You can always have to getScreen() methods. One that takes levels and goes

```
if source is null
    // initial load case, we don't want a transition screen for the target dim there
    return getSource(null, null)
else if target level is null
    // weird case, but better than an NPE, I guess...
    return getScreen(source.dim, null)
else
    return getScreen(source.dim, target.dim)
```

and the existing one.